plugins/Lv2Effect/Lv2Effect.cpp

```diff
@@ -71,26 +65,30 @@ Lv2Effect::~Lv2Effect()
 bool Lv2Effect::processAudioBuffer(sampleFrame *buf, const fpp_t frames)
 {
 	if (!isEnabled() || !isRunning()) { return false; }
+	Q_ASSERT(frames <= (fpp_t)tmpOutputSmps.size());
 
-	Lv2FxControls& ctrl = m_controls;
-
-	ctrl.copyBuffersFromLmms(buf, frames);
+	m_controls.copyBuffersFromLmms(buf, frames);
 	m_controls.copyModelsFromLmms();
 
 //	m_pluginMutex.lock();
-	ctrl.run(frames);
+	m_controls.run(frames);
 //	m_pluginMutex.unlock();
 
-	ctrl.copyBuffersToLmms(buf, frames);
+	m_controls.copyBuffersToLmms(tmpOutputSmps.data(), frames);
 
 	double outSum = .0;
+	bool corrupt = wetLevel() < 0; // #3261 - if w < 0, bash w := 0, d := 1
+	const float d = corrupt ? 1 : dryLevel();
+	const float w = corrupt ? 0 : wetLevel();
 	for(fpp_t f = 0; f < frames; ++f)
 	{
-		double l = static_cast<double>(buf[f][0]);
-		double r = static_cast<double>(buf[f][1]);
+		double l = static_cast<double>(tmpOutputSmps[f][0]);
+		double r = static_cast<double>(tmpOutputSmps[f][1]);
 		outSum += l*l + r*r;
+		buf[f][0] = d * buf[f][0] + w * tmpOutputSmps[f][0];
+		buf[f][1] = d * buf[f][1] + w * tmpOutputSmps[f][1];
```

Comment:
In the other effects, `outSum` is computed using the combined wet and dry signal (that is, using `buf[f][0]` and `buf[f][1]` after mixing). I think this is necessary in order to avoid incorrectly shutting off the audio in the case that the effect output is silent, but a non-silent dry signal is still allowed through by the wet/dry setting.

Reply:
> In the other effects, `outSum` is computed using the combined wet and dry signal

What do you mean? `outSum` is computed by combining the wet (`tmpOutputSmps[f][0]`) and dry (`buf[f][0]`) signal. What do you mean with "after mixing"?

Reply:
Currently, `outSum` is only calculated from `tmpOutputSmps`. I think the code should be
```cpp
buf[f][0] = d * buf[f][0] + w * tmpOutputSmps[f][0];
buf[f][1] = d * buf[f][1] + w * tmpOutputSmps[f][1];
outSum += buf[f][0] * buf[f][0] + buf[f][1] * buf[f][1];
```

Reply:
I believe [as seen here](https://github.com/LMMS/lmms/pull/5475), yes `outSum` should be after.

Reply:
Oh ofc, how could I not see this :smiley: